src/dbnode/encoding/proto/README.md

```diff
@@ -0,0 +1,269 @@
+# Protobuf Encoding
+
+## Overview
+
+This package contains the encoder/decoder(iterator) for compressing streams of Protobuf messages matching a provided schema.
+All compression is performed in a streaming manner such that we update the encoded stream with each write; there is no internal buffering or batching during which we gather multiple writes before performing encoding.
+
+## Features
+
+1. Compress the timestamps of the ProtoBuf messages using [Gorilla-style delta of delta encoding](https://www.vldb.org/pvldb/vol8/p1816-teller.pdf).
+2. Compress streams of ProtoBuf messages matching a provided schema by using different forms of compression for each field based on its type.
+3. Support changing the schema of a ProtoBuf message mid-stream (pending).
+4. Absolutely no lossiness for any of the compression types.
+
+## Supported Syntax
+
+This package strives to support compressing any valid Protobuf messages as specified using [Protobuf syntax version 3](https://developers.google.com/protocol-buffers/docs/proto3), however, only the following features have been tested:
+
+1. [All scalar value types](https://developers.google.com/protocol-buffers/docs/proto3#scalar)
+2. Nested Messages
+3. Repeated fields
+4. Map fields
+5. Reserved fields
+
+We have explicitly not performed any testing of the following features and thus they are not supported:
```

Comment:
Hm, I thought you were just going to allow these and not compress?

Reply:
I don't do anything to stop people from using these things and they probably work, but I haven't included them in my prop test just because I haven't had time so kind of a "use at your own risk" right now and we may just reject them outright in the schema management APIs

Nothing stopping us from supporting them later, just haven't had time to test basically